mm2src/coins/eth/eth_hd_wallet.rs

```diff
@@ -0,0 +1,206 @@
+use super::*;
+use crate::coin_balance::HDAddressBalanceScanner;
+use crate::hd_wallet::{ExtractExtendedPubkey, HDAccount, HDAddress, HDExtractPubkeyError, HDWallet, HDXPubExtractor,
+                       TrezorCoinError};
+use async_trait::async_trait;
+use bip32::DerivationPath;
+use crypto::Secp256k1ExtendedPublicKey;
+use ethereum_types::{Address, Public};
+use std::iter;
+
+pub type EthHDAddress = HDAddress<Address, Public>;
+pub type EthHDAccount = HDAccount<EthHDAddress>;
+pub type EthHDWallet = HDWallet<EthHDAccount>;
+
+#[async_trait]
+impl ExtractExtendedPubkey for EthCoin {
+    type ExtendedPublicKey = Secp256k1ExtendedPublicKey;
+
+    async fn extract_extended_pubkey<XPubExtractor>(
+        &self,
+        xpub_extractor: Option<XPubExtractor>,
+        derivation_path: DerivationPath,
+    ) -> MmResult<Self::ExtendedPublicKey, HDExtractPubkeyError>
+    where
+        XPubExtractor: HDXPubExtractor + Send,
+    {
+        extract_extended_pubkey(self, xpub_extractor, derivation_path).await
```

Comment:
maybe good to rename this func to extract_extended_pubkey_impl() or extract_extended_pubkey_common_impl()

Reply:
Fixed here https://github.com/KomodoPlatform/komodo-defi-framework/pull/1979/commits/fb1ccd724af5ae2c6ec5fec672883c131d0441a5